site/source/docs/api_reference/emscripten.h.rst

```diff
@@ -931,6 +931,15 @@ Functions
 	:returns: A pointer to the preloaded image or NULL.
 	:rtype: char*
 
+.. c:function:: char *emscripten_print_double(double x, char *to)
+
+	Prints a double as a string. This is useful because JS engines have good support for printing out a double in a way that takes the least possible size, but preserves all the information in the double, i.e., it can then be parsed back in a perfectly reversible manner (snprintf etc. do not do so, sadly).
+
+	No zero terminator is emitted, just the double itself.
```

Comment:
I think we should always write the zero terminator, even when/if the output is truncated.